src/celeritas/Types.hh

```diff
@@ -66,6 +66,9 @@ using ElementComponentId = OpaqueId<struct MatElementComponent>;
 //! Opaque index to one isotopic component datum in a particular element
 using IsotopeComponentId = OpaqueId<struct ElIsotopeComponent>;
 
+//! Opaque index to a material with optical properties
+using OpticalMaterialId = OpaqueId<struct OpticalMaterial_>;
```

Comment:
How often do different materials share optical properties? (Our current "material IDs" really should be "physics material IDs" since it incorporates the range cutoff info like "material cut couple" in Geant4. If this is really just the pure material IDs maybe it's time to rename our material ID and use this instead.)

Reply:
I was under the impression that only a subset of materials might have optical properties attached, so we would only want to store the grids for those materials. This isn't really necessary for this PR though, so I could just use `MaterialId` and we could think about this later (when we actually start adding the optical property data)...